src/plugins/chart_expressions/expression_metric/common/expression_functions/metric_vis_function.ts

```diff
@@ -45,121 +26,108 @@ export const metricVisFunction = (): MetricVisExpressionFunctionDefinition => ({
-      types: ['boolean'],
-      default: false,
-      help: i18n.translate('expressionMetricVis.function.percentageMode.help', {
-        defaultMessage: 'Shows metric in percentage mode. Requires colorRange to be set.',
+    metric: {
+      types: ['vis_dimension', 'string'],
+      help: i18n.translate('expressionMetricVis.function.metric.help', {
+        defaultMessage: 'The primary metric.',
       }),
     },
-    colorMode: {
-      types: ['string'],
-      default: `"${ColorMode.None}"`,
-      options: [ColorMode.None, ColorMode.Labels, ColorMode.Background],
-      help: i18n.translate('expressionMetricVis.function.colorMode.help', {
-        defaultMessage: 'Which part of metric to color',
+    secondaryMetric: {
+      types: ['vis_dimension', 'string'],
+      help: i18n.translate('expressionMetricVis.function.secondaryMetric.help', {
+        defaultMessage: 'The secondary metric (shown above the primary).',
       }),
-      strict: true,
     },
-    colorFullBackground: {
-      types: ['boolean'],
-      default: false,
-      help: i18n.translate('expressionMetricVis.function.colorFullBackground.help', {
-        defaultMessage: 'Applies the selected background color to the full visualization container',
+    breakdownBy: {
+      types: ['vis_dimension', 'string'],
+      help: i18n.translate('expressionMetricVis.function.breakdownBy.help', {
+        defaultMessage: 'The dimension containing the labels for sub-categories.',
       }),
     },
-    palette: {
-      types: ['palette'],
-      help: i18n.translate('expressionMetricVis.function.palette.help', {
-        defaultMessage: 'Provides colors for the values, based on the bounds.',
+    subtitle: {
+      types: ['string'],
+      help: i18n.translate('expressionMetricVis.function.subtitle.help', {
+        defaultMessage: 'The subtitle for a single metric. Overridden if breakdownBy is supplied.',
       }),
     },
-    showLabels: {
-      types: ['boolean'],
-      default: true,
-      help: i18n.translate('expressionMetricVis.function.showLabels.help', {
-        defaultMessage: 'Shows labels under the metric values.',
+    extraText: {
+      types: ['string'],
+      help: i18n.translate('expressionMetricVis.function.extra.help', {
+        defaultMessage: 'Text to be shown above metric value. Overridden by secondaryMetric.',
       }),
     },
-    font: {
-      types: ['style'],
-      help: i18n.translate('expressionMetricVis.function.font.help', {
-        defaultMessage: 'Font settings.',
+    progressMin: {
+      types: ['number'],
+      help: i18n.translate('expressionMetricVis.function.progressMin.help', {
+        defaultMessage: 'The number at which the progress bar should be empty.',
       }),
-      default: `{font size=60 align="center"}`,
     },
-    labelFont: {
-      types: ['style'],
-      help: i18n.translate('expressionMetricVis.function.labelFont.help', {
-        defaultMessage: 'Label font settings.',
+    progressMax: {
```

Comment:
As it will be possible to dynamically define the "max" via dimension in Lens, I think we need to turn this into an accessor like primary and secondary metric like it's done for gauge

Reply:
Will it still be useful to use a static value as well?

Reply:
Never mind, I see from gauges that Lens adds the static value as a data table column.